src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -3099,7 +3115,13 @@ public static ForkJoinPool commonPool() {
     public <T> T invoke(ForkJoinTask<T> task) {
         Objects.requireNonNull(task);
         poolSubmit(true, task);
-        return task.join();
+        try {
+            return task.join();
+        } catch (RuntimeException | Error unchecked) {
+            throw unchecked;
+        } catch (Exception checked) {
+            throw new RuntimeException(checked);
```

Comment:
We should probably a test that uses invoke(adaptInterruptible(callable)) to check that a checked exception is wrapped, just in case it breaks with some future change.

Reply:
Yes, somehow a test for this has been missing since first release. I added it to tck/ForkJoinPoolTest.java